pkg/controller/operands/operandHandler.go

```diff
@@ -178,7 +179,7 @@ func (h OperandHandler) EnsureDeleted(req *common.HcoRequest) error {
 				errorCh <- err
 			} else {
 				key := client.ObjectKeyFromObject(o)
-				if err := h.client.Get(tCtx, key, o); err == nil {
+				if err := h.client.Get(tCtx, key, o); apierrors.IsNotFound(err) {
```

Comment:
Adding a test case with `FakeReadErrorGenerator` + IsNotFound error may be useful.

Reply:
I tend to thik that this check is wrong. This will work with the fake client, but in the eventual consistance system like K8s, we can't prommise that the resource was already removed. I'll remove this check.. @tiraboschi - what do you think?

Reply:
hcoutil.EnsureDeleted implements a wait parameter and on line 164 we are calling it with wait=true.
In that case, we are also waiting for the resource to be really disappeared and not simply flagged for deletion so I think that the check is valid.

Reply:
This is checked in `util.validateDeletion` that called from `util.ComponentResourceRemoval`. At this place, the resource is deleted for sure

Reply:
yes, right, and if util.validateDeletion failed to delete withing the allowed time, it should return an error that is propagated and then returned from hcoutil.EnsureDeleted so for sure we are not in this branch

Reply:
Removed. No need to test it.